common/websocket/index.ts

```diff
@@ -0,0 +1,115 @@
+import { Server } from 'http';
+import { getLogger } from 'log4js';
+import { WebSocketServer, WebSocket } from 'ws';
+
+// can also be any kind of authkey
+type UserId = string;
+
+type ConnectedClients = {
+    [k: UserId]: WebSocket;
+};
+
+type Message = {
+    notificationId: string;
+};
+
+const log = getLogger();
+
+class WebSocketService {
+    private static instance: WebSocketService;
+    private websocketServer: WebSocketServer;
+    private connectedClients: ConnectedClients = {};
+
+    constructor(server: Server) {
+        if (WebSocketService.instance) {
+            return WebSocketService.instance;
+        }
+        this.websocketServer = new WebSocketServer({ server, clientTracking: true });
+    }
+
+    public static getInstance(server?: Server) {
+        if (!this.instance && !server) {
+            throw new Error('No instance running and no instanciation possible without a server.');
+        }
+        if (!this.instance) {
+            this.instance = new WebSocketService(server);
+        }
+        return this.instance;
+    }
+
+    getCients(): ConnectedClients {
```

Comment:
would it make sense to have an array of ids as optional parameter to only receive connected Clients which will get a notification? than we might not need isClientOnline method

Reply:
I think I'm not getting it, you should only get connected clients with this method. 😄